core/chains/evm/txmgr/tracker_test.go

```diff
@@ -81,11 +80,12 @@ func TestEvmTracker_AddressTracking(t *testing.T) {
 			require.NoError(t, err)
 		}(tracker)
 
-		time.Sleep(waitTime)
-		addrs := tracker.GetAbandonedAddresses()
-		require.NotContains(t, addrs, inProgressAddr)
-		require.NotContains(t, addrs, unstartedAddr)
-		require.Contains(t, addrs, unconfirmedAddr)
+		require.Eventually(t, func() bool {
+			addrs := tracker.GetAbandonedAddresses()
+			return !slices.Contains(addrs, inProgressAddr) &&
+				!slices.Contains(addrs, unstartedAddr) &&
+				slices.Contains(addrs, unconfirmedAddr)
+		}, 10*time.Second, time.Millisecond)
```

Comment:
I assume this test is expected to pass pretty quickly since the previous wait time was just 5ms. But could checking every millisecond be excessive if this runs long for 10 seconds?

Reply:
Fair point, I can change this to maybe 100ms instead.